web/src/main/java/it/pagopa/selfcare/dashboard/web/controller/UserController.java

```diff
@@ -0,0 +1,39 @@
+package it.pagopa.selfcare.dashboard.web.controller;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import it.pagopa.selfcare.dashboard.connector.model.user.User;
+import it.pagopa.selfcare.dashboard.core.UserRegistryService;
+import it.pagopa.selfcare.dashboard.web.model.EmbeddedExternalIdDto;
+import it.pagopa.selfcare.dashboard.web.model.UserResource;
+import it.pagopa.selfcare.dashboard.web.model.mapper.UserMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+@Slf4j
+@RestController
+@RequestMapping(value = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
+@Api(tags = "user")
+public class UserController {
+
+    private final UserRegistryService userRegistryService;
+
+    @Autowired
+    public UserController(UserRegistryService userRegistryService) {
+        this.userRegistryService = userRegistryService;
+    }
+
+    @PostMapping(value = "/external-id")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "", notes = "${swagger.dashboard.user.api.getUserByExternalId}")
+    public UserResource getUserByExternalId(@ApiParam()
+                                            @RequestBody
+                                                    EmbeddedExternalIdDto externalId) {
+        User user = userRegistryService.getUser(externalId.getExternalId());
+        return UserMapper.toUserResource(user);
```

Comment:
@KevinSi96 add trace and debug log. Remember to log externalId only if TargetEnv is not PROD